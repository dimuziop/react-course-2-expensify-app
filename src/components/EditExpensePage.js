import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from "./ExpenseForm";
import {startEditExpense, startRemoveExpense} from "../actions/expenses";

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.onRemove({id: this.props.expense.id});
        this.props.history.push('/')
    };

    render() {
        return (

            <div className={'dashboard-wrapper'}>
                <div className={'page-header'}>
                    <div className={'content-container'}>
                        <h1 className={'page-header__title'}>Edit Expense</h1>
                    </div>
                </div>
                <div className={"content-container"}>
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <div>
                        <button
                            className={"delete-button "}
                            onClick={this.onRemove}>
                            Delete
                        </button>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id))
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        onRemove: (data) => dispatch(startRemoveExpense({id: data.id}))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

/*
const EditExpensePage = (props) => {
    return (
        <div>
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense) => {
                    props.dispatch(editExpense(props.expense.id, expense));
                    props.history.push('/');
                }}
            />
            <button onClick={() => {
                props.dispatch
                (removeExpense
                ({id: props.expense.id}));
                props.history.push('/')
            }}>
                Delete
            </button>
        </div>
    );
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id))
    }
};

export default connect(mapStateToProps)(EditExpensePage);*/
