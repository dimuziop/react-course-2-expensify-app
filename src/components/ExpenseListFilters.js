import React from 'react';
import {connect} from 'react-redux';
import {setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate} from "../actions/filters";
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    onSortChange = (e) => {
        if (e.target.value === 'amount') {
            this.props.sortByAmount()
        } else {
            this.props.sortByDate()
        }
    };

    render() {
        return (
            <div className={'content-container'}>
                <div className={"input-group"}>
                    <div className={"input-group__item"}>
                        <input placeholder={"Search expenses"} className={"text-input"} type="text" defaultValue={this.props.filters.text} onChange={this.onTextChange}/>
                    </div>
                    <div className={"input-group__item"}>
                        <select className={"select"} value={this.props.filters.sortBy} onChange={this.onSortChange}>
                            <option value={'date'}>Date</option>
                            <option value={'amount'}>Amount</option>
                        </select>
                    </div>
                    <div className={"input-group__item"}>
                        <DateRangePicker
                            startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                            endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                            onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate())
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);