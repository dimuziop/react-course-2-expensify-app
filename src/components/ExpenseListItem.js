import React from 'react';
import {removeExpense} from "../actions/expenses";
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


export const ExpenseListItem = ({id, dispatch, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{numeral(amount/100).format('$0,0.00')} - {moment(createdAt).format('d/MM/YYYY')} </p>
        <button onClick={() => (dispatch(removeExpense({id: id})))}>Delete</button>
    </div>
);


export default connect()(ExpenseListItem);
/*
export default (props) => {

    const redered = props.expenses.map((expense) => {
        return (
            <div>
                <h3>{expense.description}</h3>
                <p>{expense.description} -- Amount: {expense.amount}</p>
            </div>
        )
    });
    return redered;
}*/
