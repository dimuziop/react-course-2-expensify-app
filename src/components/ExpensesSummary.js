import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import TotalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div className={"page-header"}>
        <div className={"content-container"}>
            {props.expenses.length === 0 ? null : (<h1 className={"page_header__title"}>Viewing <span>{props.expenses.length}</span> -
                Totalling <span className={"span"}>{numeral(TotalExpenses(props.expenses) / 100).format('$0,0.00')}</span> </h1>)}
            <div className={"page-header__actions"}>
            <Link className={"add-button ripple"} to={"/create"}>
                Add Expense
                </Link>
                </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);