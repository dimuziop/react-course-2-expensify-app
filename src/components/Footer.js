import React from 'react';

const Footer = () => (
    <header className={"footer"}>
            <p className={"footer_paragraph"}>Hello! Test API - Patricio Di Muzio</p>
    </header>
);

export default Footer;