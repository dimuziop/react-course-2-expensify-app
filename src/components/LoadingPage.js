import React from 'react';

const LoadingPage = () => (
    <div className={"loader"}>
        <img className={"loader__image"} src="/images/loader.gif" alt=""/>
        <h3 className={"loader__text"}>Loading ...</h3>
    </div>
);

export default LoadingPage;