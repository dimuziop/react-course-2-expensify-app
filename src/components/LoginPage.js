import React from 'react';
import { connect } from 'react-redux';
import {startLogin} from "../actions/auth";

export const LoginPage = ( {startLogin} ) => (
    <div className={"box-layout"}>
        <div className={"box-layout__box"}>
            <p>It's time to get your expenses under control</p>
        <button className={"login-button ripple"} onClick={ startLogin }>
            Login with Google
        </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);