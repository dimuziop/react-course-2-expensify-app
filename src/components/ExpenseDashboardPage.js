import React from 'react';
import ConectedExpenseList from './ExpenseList';
import ExpenseListFilters from "./ExpenseListFilters";


const ExpenseDashboardPage = (props) => (
    <div>
        <ExpenseListFilters />
        <ConectedExpenseList />
    </div>
);

export default ExpenseDashboardPage;