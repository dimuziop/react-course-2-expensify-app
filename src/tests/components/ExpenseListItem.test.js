import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render a single item', () => {
    const wrapper = shallow(<ExpenseListItem key={expenses[1].id} {...expenses[1]}/>);
    expect(wrapper).toMatchSnapshot();
});