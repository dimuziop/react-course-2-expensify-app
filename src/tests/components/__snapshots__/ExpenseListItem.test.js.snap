// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Test2
    </h3>
  </Link>
  <p>
    $50.00
     - 
    6/12/1969
     
  </p>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
